rules/group.go

```diff
@@ -111,21 +117,27 @@ func NewGroup(o GroupOptions) *Group {
 		evalIterationFunc = DefaultEvalIterationFunc
 	}
 
+	concurrencyController := o.Opts.RuleConcurrencyController
+	if concurrencyController == nil {
```

Comment:
Note to reviewers: this is useful for unit tests, to having having to explicitly set it everywhere.